test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
wait.Poll also doesn't allow you to return the error you got so this is going to be a mess (close around an error variable also, lie about not getting an error to the poller until it times out, then check if the poller timed out or succeeded and return the response variable or the error variable).  I'm not seeing the benefit in changing it.